pageserver/src/metrics.rs

```diff
@@ -548,7 +548,52 @@ const STORAGE_IO_TIME_BUCKETS: &[f64] = &[
 /// - seek (modify internal position or file length query)
 /// - fsync ([`std::fs::File::sync_all`])
 /// - metadata ([`std::fs::File::metadata`])
-pub(crate) static STORAGE_IO_TIME: Lazy<HistogramVec> = Lazy::new(|| {
+#[derive(
+    Debug, Clone, Copy, strum_macros::EnumCount, strum_macros::EnumIter, strum_macros::FromRepr,
+)]
+pub(crate) enum StorageIoOperation {
+    Open,
+    Close,
+    CloseByReplace,
+    Read,
+    Write,
+    Seek,
+    Fsync,
+    Metadata,
+}
+
+impl StorageIoOperation {
```

Comment:
Why not to use use `strum_macros::EnumString` ?

Reply:
Um, afaik, this [CloseByReplace](https://github.com/duguorong009/neon/blob/gr/storage-io-time-metrics-improval/pageserver/src/metrics.rs#L570C70-L570C70) cannot be directly built from `string`.
That's why I added the manual impls.